gobblin-api/src/main/java/org/apache/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -496,6 +496,7 @@
   public static final String DATA_PUBLISHER_OVERWRITE_ENABLED = DATA_PUBLISHER_PREFIX + ".overwrite.enabled";
   // This property is used to specify the owner group of the data publisher final output directory
   public static final String DATA_PUBLISHER_FINAL_DIR_GROUP = DATA_PUBLISHER_PREFIX + ".final.dir.group";
+  public static final String DATA_PUBLISHER_OUTPUT_DIR_GROUP = DATA_PUBLISHER_PREFIX + ".output.dir.group";
```

Comment:
What's the difference between "final" and "output" dir for publisher?

Reply:
Although the FINAL DIR and OUTPUT DIR are essentially the same dir (/db/table in case of databases) but the DATA_PUBLISHER_FINAL_DIR_GROUP is applied at the leaf level (/db/table/yyyy/mm/dd/hh) while DATA_PUBLISHER_OUTPUT_DIR_GROUP is applied at the table level (/db/table which is what we want)

Reply:
"writer.group.name" is applied to the actual data file (avro, orc) while "publisher.final.dir.group" only applied the group at leaf level (DATA_PUBLISHER_FINAL_DIR_GROUP)

Reply:
This is confusing. If FINAL DIR and OUTPUT DIR are the same, why does the group apply to different folders?

Also, what group will apply to folder at intermediate level, like "/db/table/yyyy/mm" ?

Reply:
Discussed over the chat. The summary is that the old setting works incorrectly, but even if it would work correctly, it would not be what we need.

From vbohra:
so the publisherOutputDir can be the following based on use case
- data.publisher.final.dir ( if data.publisher.appendExtractToFinalDir is set to false)
- data.publisher.final.dir/db/table ( if data.publisher.appendExtractToFinalDir is set to true and writer.file.path.type = namespace_table) [ THIS IS THE CASE FOR brooklin-etl ]
- data.publisher.final.dir/table ( if data.publisher.appendExtractToFinalDir is set to true and writer.file.path.type = tablename)
- and a default